HomeWork/HomeWork/UserService.cs

```diff
@@ -0,0 +1,578 @@
+
+            Assert.AreEqual(textItem, pageWithItem, "Wrong!!");
+        }
+
+        public void PriceFilter()
+        {
+            Actions actions = new Actions(driver);
+
+            var gadjetItem = driver.FindElement(_gadjetItemButton);
+            actions.MoveToElement(gadjetItem).Perform();
+
+            Thread.Sleep(1000);
+
+            var gadjetDropButton = driver.FindElement(_gadjetDropButton);
+            gadjetDropButton.Click();
+
+            var filterMobileBrand = driver.FindElement(_moBileBrandFilterButton);
+            filterMobileBrand.Click();
+
+            Thread.Sleep(1000);
+
+            var showFilter = driver.FindElement(_showFilter);
+            showFilter.Click();
+
+            Thread.Sleep(2000);
+
+            var itemPro = driver.FindElement(_proItemApple);
+            // actions.MoveToElement(itemPro).Perform();
+            itemPro.Click();
+
+            var showAllPr = driver.FindElement(_showAllPriceButton);
+            showAllPr.Click();
+
+            Thread.Sleep(2000);
+
+            var sortPrice = driver.FindElement(_sortPriceOnPageButton);
+            sortPrice.Click();
+
+            Thread.Sleep(2000);
+
+            // var listAllPrice = driver.FindElements(_priceAllItem);
+        }
+
+        public void AddToBookmarks()
+        {
+            Actions actions = new Actions(driver);
+
+            var gadjetMenu = driver.FindElement(_gadjetItemButton);
+            actions.MoveToElement(gadjetMenu).Perform();
+
+            Thread.Sleep(1000);
+
+            var mobileMenu = driver.FindElement(_gadjetDropButton);
+            mobileMenu.Click();
+
+            var filterMobileBrand = driver.FindElement(_moBileBrandFilterButton);
+            filterMobileBrand.Click();
+
+            Thread.Sleep(1000);
+
+            var showFilter = driver.FindElement(_showFilter);
+            showFilter.Click();
+
+            var appleMobItem = driver.FindElement(_appleMobileItem);
+            appleMobItem.Click();
+
+            var nameTitleItem = driver.FindElement(By.XPath("//h1[@class='t2 no-mobile' and text()='Мобильный телефон Apple iPhone 13 ']")).Text;
+
+            Thread.Sleep(2000);
+
+            var addToBookmarks = driver.FindElement(_addToBookmarksButton);
+            addToBookmarks.Click();
+
+            Thread.Sleep(1000);
+
+            var bookMarksBut = driver.FindElement(_addToBookmarksButton);
+            bookMarksBut.Click();
+
+            Thread.Sleep(1000);
+
+            var bookmarksBut = driver.FindElement(_bookmarksButton);
+            bookmarksBut.Click();
+
+            Thread.Sleep(2000);
+
+            var textItemInBookmarks = driver.FindElement(By.XPath("//div[@class='side-list-label ' and text()='Apple iPhone 13 128GB']")).Text;
+            var nameItemInBooksmarks = textItemInBookmarks.Replace("GB", "");
+
+            Assert.IsTrue(nameTitleItem.Contains(nameItemInBooksmarks), "Wrong");
```

Comment:
nice error message, if I would see that one, I'm absolute will understand what's wrong :)